/*
 * This file was automatically generated by EvoSuite
 * Tue Feb 19 15:53:49 GMT 2019
 */

package org.sat4j.minisat.orders;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import org.sat4j.minisat.constraints.CardinalityDataStructure;
import org.sat4j.minisat.constraints.cnf.Lits;
import org.sat4j.minisat.core.ILits;
import org.sat4j.minisat.orders.SubsetVarOrder;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class SubsetVarOrder_ESTest extends SubsetVarOrder_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      int[] intArray0 = new int[8];
      intArray0[0] = 916;
      SubsetVarOrder subsetVarOrder0 = new SubsetVarOrder(intArray0);
      CardinalityDataStructure cardinalityDataStructure0 = new CardinalityDataStructure();
      ILits iLits0 = cardinalityDataStructure0.getVocabulary();
      iLits0.ensurePool(1000);
      subsetVarOrder0.setLits(iLits0);
      // Undeclared exception!
      try { 
        subsetVarOrder0.init();
        fail("Expecting exception: AssertionError");
      
      } catch(AssertionError e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      CardinalityDataStructure cardinalityDataStructure0 = new CardinalityDataStructure();
      ILits iLits0 = cardinalityDataStructure0.getVocabulary();
      int[] intArray0 = new int[4];
      intArray0[0] = (-1379);
      SubsetVarOrder subsetVarOrder0 = new SubsetVarOrder(intArray0);
      subsetVarOrder0.setLits(iLits0);
      // Undeclared exception!
      try { 
        subsetVarOrder0.init();
        fail("Expecting exception: AssertionError");
      
      } catch(AssertionError e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      int[] intArray0 = new int[2];
      SubsetVarOrder subsetVarOrder0 = new SubsetVarOrder(intArray0);
      double[] doubleArray0 = new double[2];
      subsetVarOrder0.activity = doubleArray0;
      CardinalityDataStructure cardinalityDataStructure0 = new CardinalityDataStructure();
      ILits iLits0 = cardinalityDataStructure0.getVocabulary();
      subsetVarOrder0.setLits(iLits0);
      // Undeclared exception!
      try { 
        subsetVarOrder0.init();
        fail("Expecting exception: AssertionError");
      
      } catch(AssertionError e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      int[] intArray0 = new int[1];
      SubsetVarOrder subsetVarOrder0 = new SubsetVarOrder(intArray0);
      // Undeclared exception!
      try { 
        subsetVarOrder0.undo((-1073));
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.sat4j.minisat.orders.SubsetVarOrder", e);
      }
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      int[] intArray0 = new int[0];
      SubsetVarOrder subsetVarOrder0 = new SubsetVarOrder(intArray0);
      Lits lits0 = new Lits();
      subsetVarOrder0.setLits(lits0);
      subsetVarOrder0.init();
      // Undeclared exception!
      try { 
        subsetVarOrder0.undo(1);
        fail("Expecting exception: ArrayIndexOutOfBoundsException");
      
      } catch(ArrayIndexOutOfBoundsException e) {
         //
         // 1
         //
         verifyException("org.sat4j.minisat.orders.SubsetVarOrder", e);
      }
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      int[] intArray0 = new int[0];
      SubsetVarOrder subsetVarOrder0 = new SubsetVarOrder(intArray0);
      Lits lits0 = new Lits();
      lits0.getFromPool(16384);
      subsetVarOrder0.setLits(lits0);
      // Undeclared exception!
      subsetVarOrder0.init();
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      int[] intArray0 = new int[0];
      SubsetVarOrder subsetVarOrder0 = new SubsetVarOrder(intArray0);
      // Undeclared exception!
      try { 
        subsetVarOrder0.init();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.sat4j.minisat.orders.SubsetVarOrder", e);
      }
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      SubsetVarOrder subsetVarOrder0 = null;
      try {
        subsetVarOrder0 = new SubsetVarOrder((int[]) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.sat4j.minisat.orders.SubsetVarOrder", e);
      }
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      int[] intArray0 = new int[0];
      SubsetVarOrder subsetVarOrder0 = new SubsetVarOrder(intArray0);
      Lits lits0 = new Lits();
      subsetVarOrder0.setLits(lits0);
      subsetVarOrder0.init();
      subsetVarOrder0.undo(0);
      assertEquals(0, subsetVarOrder0.numberOfInterestingVariables());
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      int[] intArray0 = new int[1];
      intArray0[0] = 6;
      SubsetVarOrder subsetVarOrder0 = new SubsetVarOrder(intArray0);
      Lits lits0 = new Lits();
      lits0.getFromPool(6);
      subsetVarOrder0.setLits(lits0);
      subsetVarOrder0.init();
      assertEquals(0, subsetVarOrder0.numberOfInterestingVariables());
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      int[] intArray0 = new int[1];
      intArray0[0] = 36;
      SubsetVarOrder subsetVarOrder0 = new SubsetVarOrder(intArray0);
      Lits lits0 = new Lits();
      subsetVarOrder0.setLits(lits0);
      // Undeclared exception!
      try { 
        subsetVarOrder0.init();
        fail("Expecting exception: AssertionError");
      
      } catch(AssertionError e) {
         //
         // 0/36
         //
      }
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      int[] intArray0 = new int[1];
      intArray0[0] = 19;
      SubsetVarOrder subsetVarOrder0 = new SubsetVarOrder(intArray0);
      Lits lits0 = new Lits();
      lits0.ensurePool(19);
      subsetVarOrder0.setLits(lits0);
      subsetVarOrder0.init();
      subsetVarOrder0.undo(19);
      subsetVarOrder0.undo(19);
      assertEquals(0, subsetVarOrder0.numberOfInterestingVariables());
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      int[] intArray0 = new int[4];
      SubsetVarOrder subsetVarOrder0 = new SubsetVarOrder(intArray0);
      subsetVarOrder0.activity = null;
      Lits lits0 = new Lits();
      subsetVarOrder0.setLits(lits0);
      // Undeclared exception!
      try { 
        subsetVarOrder0.init();
        fail("Expecting exception: AssertionError");
      
      } catch(AssertionError e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }
}
